CRM/Core/Form.php

```diff
@@ -1177,8 +1177,8 @@ public function addField($name, $props = array(), $required = FALSE) {
     // Get field metadata.
     $fieldSpec = civicrm_api3($props['entity'], 'getfield', $props);
     $fieldSpec = $fieldSpec['values'];
-
-    $label = CRM_Utils_Array::value('label', $props, $fieldSpec['title']);
+    $label = CRM_Utils_Array::value('label', $props, isset($fieldSpec['title']) ? $fieldSpec['title'] : NULL);
+    $name = isset($props['name_id']) ? $props['name_id'] : $name;
```

Comment:
Maybe I'm not understanding this exactly, but it seems like this might be redundant with the opposite approach of specifying `$props['name']` and passing a different value for `$name`. So in the example below, instead of

    $this->addField('rule_length', array('entity' => 'Rule', 'name_id' => "length_$count")

the equivalent (which wouldn't require adding the extra name_id param to this function) would be

    $this->addField("length_$count", array('entity' => 'Rule', 'name' => 'rule_length')